docs/proposals/202001_thanos_query_health_handling.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Thanos Query store nodes healthiness handling
+type: proposal
+menu: proposals
+status: proposed
+owner: GiedriusS
+---
+
+### Related Tickets
+
+* https://github.com/thanos-io/thanos/issues/1651 (Response caching for Thanos)
+* https://github.com/thanos-io/thanos/pull/2072 (query: add 'sticky' nodes)
+* https://github.com/cortexproject/cortex/pull/1974 (frontend: implement cache control)
+* https://github.com/thanos-io/thanos/pull/1984 (api/v1: add cache-control header)
+
+## Summary
+
+This proposal document describes how currently the healthiness of store nodes is handled and why it should be changed to work better in terms of end response caching.
+
+It explores a few options that are available - weighs their pros and cons, and finally proposes one final variant which seems to be the best out of the possible ones.
```

Comment:
```suggestion
It explores a few options that are available - weighs their pros and cons, and finally proposes one final variant which we believe is the best out of the possible ones.
```

Reply:
Yep, added this as well :+1: I have used that language previously since I have decided that myself personally, and now we have decided to use this option so this is appropriate.